Code/GraphMol/RascalMCES/RascalMCES.cpp

```diff
@@ -0,0 +1,1086 @@
+    }
+  }
+}
+
+RascalStartPoint makeInitialPartitionSet(const ROMol *mol1, const ROMol *mol2,
+                                         const RascalOptions &opts) {
+  RascalStartPoint starter;
+  if (mol1->getNumAtoms() <= mol2->getNumAtoms()) {
+    starter.d_swapped = false;
+    starter.d_mol1 = mol1;
+    starter.d_mol2 = mol2;
+  } else {
+    starter.d_swapped = true;
+    starter.d_mol1 = mol2;
+    starter.d_mol2 = mol1;
+  }
+  std::map<int, std::vector<std::pair<int, int>>> degSeqs1, degSeqs2;
+  starter.d_tier1Sim =
+      details::tier1Sim(*starter.d_mol1, *starter.d_mol2, degSeqs1, degSeqs2);
+  if (starter.d_tier1Sim < opts.similarityThreshold) {
+    return starter;
+  }
+  std::vector<unsigned int> bondLabels1, bondLabels2;
+  details::getBondLabels(*starter.d_mol1, *starter.d_mol2, opts, bondLabels1,
+                         bondLabels2);
+  starter.d_tier2Sim =
+      details::tier2Sim(*starter.d_mol1, *starter.d_mol2, degSeqs1, degSeqs2,
+                        bondLabels1, bondLabels2);
+  if (starter.d_tier2Sim < opts.similarityThreshold) {
+    return starter;
+  }
+
+  // Get the line graphs for the two molecules as adjacency matrices.
+  makeLineGraph(*starter.d_mol1, starter.d_adjMatrix1);
+  makeLineGraph(*starter.d_mol2, starter.d_adjMatrix2);
+
+  std::vector<std::vector<int>> distMat1, distMat2;
+  if (opts.maxFragSeparation > -1 || opts.singleLargestFrag) {
+    calcDistMatrix(starter.d_adjMatrix1, distMat1);
+    calcDistMatrix(starter.d_adjMatrix2, distMat2);
+  }
+
+  // pairs are vertices in the 2 line graphs that are the same type.
+  // mod_prod is the modular product/correspondence graph of the two
+  // line graphs.
+  makeModularProduct(*starter.d_mol1, starter.d_adjMatrix1, bondLabels1,
+                     distMat1, *starter.d_mol2, starter.d_adjMatrix2,
+                     bondLabels2, distMat2, opts, starter.d_vtxPairs,
+                     starter.d_modProd);
+  if (starter.d_modProd.empty()) {
+    return starter;
+  }
+  starter.d_lowerBound = calcLowerBound(*starter.d_mol1, *starter.d_mol2,
+                                        opts.similarityThreshold);
+
+  starter.d_partSet.reset(new PartitionSet(starter.d_modProd,
+                                           starter.d_vtxPairs, bondLabels1,
+                                           bondLabels2, starter.d_lowerBound));
+
+  starter.d_deltaYPoss =
+      deltaYExchangePossible(*starter.d_mol1, *starter.d_mol2);
+
+  if (opts.doEquivBondPruning) {
+    // if equiv_bonds1[i] and equiv_bonds1[j] are equal, the bonds are
+    // equivalent.
+    findEquivalentBonds(*starter.d_mol1, starter.d_equivBonds1);
+    findEquivalentBonds(*starter.d_mol2, starter.d_equivBonds2);
+  } else {
+    starter.d_equivBonds1 = std::vector<int>(starter.d_mol1->getNumBonds(), -1);
+    starter.d_equivBonds2 = std::vector<int>(starter.d_mol2->getNumBonds(), -1);
+  }
+
+  return starter;
+}
+
+std::vector<RascalResult> findMces(RascalStartPoint &starter,
```

Comment:
it's not in the public API, so not that important, but this should be `findMCES`